lib/rubocop/cop/style/single_argument_dig.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Sometimes using dig method ends up with just a single
+      # argument. In such cases, dig should be replaced with [].
+      #
+      # @example
+      #   # bad
+      #   { key: 'value' }.dig(:key)
+      #   [1, 2, 3].dig(0)
+      #
+      #   # good
+      #   { key: 'value' }[:key]
+      #   [1, 2, 3][0]
+      #
+      #   # good
+      #   { key1: { key2: 'value' } }.dig(:key1, :key2)
+      #   [1, [2, [3]]].dig(1, 1)
+      #
+      #   # good
+      #   keys = %i[key1 key2]
+      #   { key1: { key2: 'value' } }.dig(*keys)
+      #
+      class SingleArgumentDig < Cop
+        MSG = 'Use `%<caller>s[%<argument>s]` instead of `%<original>s`.'
+
+        def_node_matcher :single_argument_dig?, <<~PATTERN
+          (send _ :dig [_ (!:splat ...)])
+        PATTERN
+
+        def on_send(node)
+          add_offense(node) if single_argument_dig?(node)
+        end
+
+        def autocorrect(node)
+          correct_access = "#{dig_caller(node)}[#{dig_argument(node)}]"
+
+          lambda do |corrector|
+            corrector.replace(node, correct_access)
+          end
+        end
+
+        private
+
+        def message(node)
+          format(
+            MSG,
+            caller: dig_caller(node),
+            argument: dig_argument(node),
+            original: node.source
+          )
+        end
+
+        def dig_caller(node)
+          node.children.first.source
```

Comment:
@marcandre I get
`undefined method `children' for #<Parser::Source::Range (string) 0...26>`
when using `Cop::Base` instead of `Cop::Cop` and running tests.
Do you have an idea?

Reply:
Well, callbacks will receive a `Source::Range` instead of `Node`, including `message`...
I'd recommend refactoring your call to `add_offsense` and specify your message then